runtime/runtime/src/bandwidth_scheduler/mod.rs

```diff
@@ -38,55 +48,89 @@ pub fn run_bandwidth_scheduler(
         Some(prev_state) => prev_state,
         None => {
             tracing::debug!(target: "runtime", "Bandwidth scheduler state not found - initializing");
-            BandwidthSchedulerState { mock_data: [0; 32] }
+            BandwidthSchedulerState {
+                link_allowances: Vec::new(),
+                sanity_check_hash: CryptoHash::default(),
+            }
         }
     };
 
-    let scheduler_state_hash: CryptoHash = hash(&borsh::to_vec(&scheduler_state).unwrap());
+    // Prepare the status info for each shard based on congestion info.
+    let mut shards_status: BTreeMap<ShardId, ShardStatus> = BTreeMap::new();
+    for (shard_id, extended_congestion_info) in apply_state.congestion_info.iter() {
+        let last_chunk_missing = extended_congestion_info.missed_chunks_count > 0;
+        let is_allowed_sender_shard =
+            *shard_id == ShardId::from(extended_congestion_info.congestion_info.allowed_shard());
```

Comment:
nit: Would be cool to make allowed shard return ShardId (not need to do it here though, unless you feel like it)

Reply:
Yeah all usages of it wrap it into ShardId anyway, let me know if you want to do it, if not I'll create a good first issue for this.

Reply:
I'd prefer to keep refactors in this PR to the minimum, it's already big enough.